src/dialogs/quick-open/ha-quick-open-dialog.ts

```diff
@@ -0,0 +1,249 @@
+import "../../components/ha-header-bar";
+import "@polymer/paper-input/paper-input";
+import "@material/mwc-list/mwc-list-item";
+import "@material/mwc-list/mwc-list";
+import {
+  css,
+  customElement,
+  html,
+  internalProperty,
+  LitElement,
+  property,
+} from "lit-element";
+import { fireEvent } from "../../common/dom/fire_event";
+import "../../components/ha-dialog";
+import { haStyleDialog } from "../../resources/styles";
+import { HomeAssistant } from "../../types";
+import { PolymerChangedEvent } from "../../polymer-types";
+import { fuzzySequentialMatch } from "../../common/string/sequence_matching";
+import { componentsWithService } from "../../common/config/components_with_service";
+import { domainIcon } from "../../common/entity/domain_icon";
+import { computeDomain } from "../../common/entity/compute_domain";
+import { domainToName } from "../../data/integration";
+import { QuickOpenDialogParams } from "./show-dialog-quick-open";
+import { HassEntity } from "home-assistant-js-websocket";
+import { compare } from "../../common/string/compare";
+import { litLocalizeLiteMixin } from "../../mixins/lit-localize-lite-mixin";
+
+interface CommandItem {
+  text: string;
+  domain: string;
+  service: string;
+  serviceData?: {};
+  onClick(ev: Event): void;
+}
+
+@customElement("ha-quick-open-dialog")
+export class QuickOpenDialog extends litLocalizeLiteMixin(LitElement) {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @internalProperty() private _commandItems: CommandItem[] = [];
+
+  @internalProperty() private _itemFilter = "";
+
+  @internalProperty() private _opened = false;
+
+  @internalProperty() private _commandMode = false;
+
+  public translationFragment = "quick-open";
+
+  public async showDialog(params: QuickOpenDialogParams) {
+    this._commandMode = params.commandMode || false;
+    this._opened = true;
+  }
+
+  public closeDialog() {
+    this._opened = false;
+    this._itemFilter = "";
+    fireEvent(this, "dialog-closed", { dialog: this.localName });
+  }
+
+  protected updated(changedParams) {
+    super.updated(changedParams);
+    if (changedParams.has("resources")) {
```

Comment:
This is the only way I could figure out how to calculate and cache the command list AFTER the strings were finished loading. The string load is async so I was caching a list of empty strings by the time we got the strings back.

Reply:
Removed the funky lazy loading of strings. Project for another day